code/js/popup.js

```diff
@@ -37,73 +37,102 @@ function dumpBookmarks(query)
 	$("#front").hide();
 
 	var found = 0;
-	var single_query = query.split(" ");
-	console.log(single_query);
+	var single = query.split(" ");
+	console.log(single);
+
+	var single_q = query.replace(/\s+/g,' ');// all words sep by ' ' 
+	var single_query = single_q.replace(/\s/g,'_');// now by '_'
+	if(single_query[0]=='_')
+		single_query=single_query.substr(1);
+	if(single_query.slice(-1)=='_')
+		single_query=single_query.slice(0,-1);// now fully by '_'
 
 	var found_word = 0;
-	var total=0;
 
-	for(var pos in single_query)
+	for(var pos in single)
 	{
 
-		current_query = single_query[pos];
+		current_query = single[pos];
 		if(current_query != "")
 		{
 			found_word = 1;
 			break;
 		}
 	}
 
+	var s,S,flag=0;
```

Comment:
You may use variable names that are more self-explanatory. 

Concerned variables: `s`, `S`, `t`

This is because, currently, we do not have any documentation. You may add a few comments (regarding the reasoning) as well.

Reply:
My bad on forgetting this point :sweat_smile: